subsys/bluetooth/host/scan.c

```diff
@@ -752,6 +798,23 @@ void bt_hci_le_per_adv_sync_established(struct net_buf *buf)
 		return;
 	}
 
+	pending_per_adv_sync->report_truncated = false;
+	if (pending_per_adv_sync->reassembly) {
+		net_buf_reset(pending_per_adv_sync->reassembly);
+	} else {
+#if CONFIG_BT_PER_ADV_SYNC_BUF_SIZE > 0
+		pending_per_adv_sync->reassembly =
+		   net_buf_alloc(&per_adv_sync_buf_pool, K_NO_WAIT);
+		if (!pending_per_adv_sync->reassembly) {
+			BT_ERR("Failed to allocate a reassembly buffer for fragmented "
+			       "advertising reports. These reports will be dropped.");
+			return;
+		}
+#else
+		pending_per_adv_sync->reassembly = NULL;
```

Comment:
If `CONFIG_BT_PER_ADV_SYNC_BUF_SIZE == 0`, shouldn't `reassembly` be guarded in the struct itself, and be fully unavailable, instead of "just" being NULL?

Reply:
The main reason for doing it this way was to avoid #ifdef spaghetti at the cost of wasting 4 bytes. I will update it if the #ifdef solution is preferred.

Reply:
I guess this will also be changed if the suggestion from https://github.com/zephyrproject-rtos/zephyr/pull/40596#discussion_r760178117 is implemented.